### DISCUSSION TITLE
How To Connect To Active Directory

### DISCUSSION THREAD

**Discussion by ghost** [coder/code-server#5081] [Q&A]
Hello,

there is some organization with already working Active Directory.

There is an idea to get some students access to the `code-server`. Required features are

1. define user as organization employee in ActiveDirectory
2. ability to filter employees to get only necessary students only.

Can someone suggest solution to the task? As I understand, there is no plugin for AD / LDAP auth.

**Comment by jsjoeio** [comment]
Hmm...I'm not sure how you would implement that. Maybe you could use [Pomerium](https://www.pomerium.com/guides/code-server.html)?

**Reply by geiseri** [reply]
[Authelia](https://www.authelia.com/) + [Traefik](https://traefik.io/traefik/) work perfectly with active directory.

**Reply by viharm** [reply]
@geiseri , Thanks for confirming that it works with _Authelia_. Does _code-server_ accept the default headers (`Remote-User`, `Remote-Groups`, etc.) forwarded by _Authelia_? Is there any other config needed?

**Reply by geiseri** [reply]
I use a user specific URL path, and auth that path per user.  Once they have the PWA saved they don't need to remember it.  I have the containers spun up manually per-user though, so you might need some sort of "helper" if you want to do it automatically.   I keep meaning to use https://plugins.traefik.io/plugins/628c9eae108ecc83915d7756/containers-on-demand to dynamically spin up a code docker instance when a user requests it and then auth it with https://plugins.traefik.io/plugins/628c9f0bffc0cd18356a97b6/path-auth.  I only have a few users, so I am okay with a small python script.  Depending on the user count you might want more automation.
